### PR TITLE
#167727324 Trip approval notification

### PR DESCRIPTION
**Pull request by joaquinto**
#### What does this PR do?

- Manager can approve an open trip

#### Description of Task to be completed?

- Create a notification model
- Installed socket.io
- Create update trip middleware
- Create approve trip request controller
- Create trip request seeder file
- Create approve trip route
- Create get trip route
- Write swagger documentations for the above routes
- Write integration test for the above route

#### Any background context you want to provide?
-  Users were not able to get notifications when their trip requests have been approved,  Hence, this PR enables staff to get notifications when their trip have been approved by their manager.

#### How should this be manually tested?

- clone the repo and cd into the root directory
- create a postgresql database
- `npm install`
- `npm run start:dev`
- use postman and test the following endpoints 
* PATCH  `/api/v1/trips/<:tripId>`
* GET  `/api/v1/trips/<:tripId>`

#### What are the relevant PT stories?
(#167727324)[https://www.pivotaltracker.com/story/show/167727324]

### CODE REVIEW

File: src/controllers/tripController.js [hunk 1]
```diff
@@ -1,6 +1,8 @@
-import * as tripServices from '../services/tripServices';
+import { postTrip, updateTripStatus, getRequesterEmail,  findTripById } from '../services/tripServices';
 import { respondWithSuccess, respondWithWarning } from '../helpers/responseHandler';
 import statusCode from '../helpers/statusCode';
+import resMessage from '../helpers/responseMessages';
```
Comment (Hector101) [inline src/controllers/tripController.js]: let's avoid creating `resMessage ` by pass the response message to the response method directly, given it's not a constant like `statusCode`.

File: src/controllers/tripController.js [hunk 2]
```diff
@@ -12,10 +14,36 @@ export const oneWayTripRequest = async (req, res) => {
...
   } catch (error) {
     return respondWithWarning(res, statusCode.internalServerError, 'Internal Server Error');
   }
 };
+
+export const approveTripRequest = async (req, res) => {
+  const status = 'approved';
+  try {
+    const tripRequest = await updateTripStatus(req.params.tripId, status);
+    const { 1: { dataValues } } = tripRequest;
```
Comment (Hector101) [inline src/controllers/tripController.js:28]: how about tripRequest.toJSON()? Also, I'd like to understand why we'd have the `dataValues` nested in a property `1`?

File: src/middlewares/tripMiddleware.js [hunk 3]
```diff
@@ -0,0 +1,20 @@
+import { findTripById } from '../services/tripServices';
+import { respondWithWarning } from '../helpers/responseHandler';
+import statusCode from '../helpers/statusCode';
+import resMessage from '../helpers/responseMessages';
+
+export const isTripExist = async (req, res, next) => {
```
Comment (Hector101) [inline src/middlewares/tripMiddleware.js]: `isTripExist ` could be renamed to `verifyTrip`

File: src/controllers/tripController.js (hunk 1 above)
Comment (joaquinto) [reply src/controllers/tripController.js]: ok. I will do that.

File: src/controllers/tripController.js (hunk 2 above)
Comment (joaquinto) [reply src/controllers/tripController.js:28]: I tried using `tripResponse.toJSON()` but it was throwing a `500` error so i decided to leave it this way

File: src/middlewares/tripMiddleware.js (hunk 3 above)
Comment (joaquinto) [reply src/middlewares/tripMiddleware.js]: ok.Thanks for the feedback. I will do that now.

**Review by Hector101 — APPROVED** [review]

**Pull request merged**